os_stub/spdm_device_secret_lib_sample/lib.c

```diff
@@ -28,6 +28,789 @@
+        }
+        *context = rsa_context;
+        return true;
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P256:
+        ec_context = libspdm_ec_new_by_nid(LIBSPDM_CRYPTO_NID_ECDSA_NIST_P256);
+        if (ec_context == NULL) {
+            return false;
+        }
+        result = libspdm_ec_set_pub_key(ec_context, m_libspdm_ec256_responder_public_key,
+                                        sizeof(m_libspdm_ec256_responder_public_key));
+        if (!result) {
+            libspdm_ec_free(ec_context);
+            return false;
+        }
+        result = libspdm_ec_set_priv_key(ec_context, m_libspdm_ec256_responder_private_key,
+                                         sizeof(m_libspdm_ec256_responder_private_key));
+        if (!result) {
+            libspdm_ec_free(ec_context);
+            return false;
+        }
+        *context = ec_context;
+        return true;
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P384:
+        ec_context = libspdm_ec_new_by_nid(LIBSPDM_CRYPTO_NID_ECDSA_NIST_P384);
+        if (ec_context == NULL) {
+            return false;
+        }
+        result = libspdm_ec_set_pub_key(ec_context, m_libspdm_ec384_responder_public_key,
+                                        sizeof(m_libspdm_ec384_responder_public_key));
+        if (!result) {
+            libspdm_ec_free(ec_context);
+            return false;
+        }
+        result = libspdm_ec_set_priv_key(ec_context, m_libspdm_ec384_responder_private_key,
+                                         sizeof(m_libspdm_ec384_responder_private_key));
+        if (!result) {
+            libspdm_ec_free(ec_context);
+            return false;
+        }
+        *context = ec_context;
+        return true;
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_ECDSA_ECC_NIST_P521:
+        ec_context = libspdm_ec_new_by_nid(LIBSPDM_CRYPTO_NID_ECDSA_NIST_P521);
+        if (ec_context == NULL) {
+            return false;
+        }
+        result = libspdm_ec_set_pub_key(ec_context, m_libspdm_ec521_responder_public_key,
+                                        sizeof(m_libspdm_ec521_responder_public_key));
+        if (!result) {
+            libspdm_ec_free(ec_context);
+            return false;
+        }
+        result = libspdm_ec_set_priv_key(ec_context, m_libspdm_ec521_responder_private_key,
+                                         sizeof(m_libspdm_ec521_responder_private_key));
+        if (!result) {
+            libspdm_ec_free(ec_context);
+            return false;
+        }
+        *context = ec_context;
+        return true;
+    default:
+        LIBSPDM_ASSERT(false);
+        return false;
+    }
+}
+
+bool libspdm_get_requester_private_key_from_raw_data(uint32_t base_asym_algo, void **context)
+{
+    bool result;
+    void *ec_context;
+    void *rsa_context;
+
+    rsa_context = NULL;
+    ec_context = NULL;
+
+    switch (base_asym_algo) {
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_2048:
```

Comment:
The logic for RSA are same. Please merge them into one common case.

The only difference is the key, you can use a sub switch for them.